app/models/user.rb

```diff
@@ -206,7 +207,42 @@ def self.authenticator(username = nil)
   end
 
   def self.authenticate(username, password, request = nil, options = {})
-    authenticator(username).authenticate(username, password, request, options)
+    user = authenticator(username).authenticate(username, password, request, options)
+    user.try(:link_to_session, request)
+
+    user
+  end
+
+  def link_to_session(request)
+    return unless request
+
+    session_id = request.session_options[:id]
```

Comment:
When this code is using this in API, then variable `session_id ` is `nil` then it  creates `Session`  record with `session_id == nil` and then it causes error in calling method `ManageIQ::Session.revoke(current_sessions.map(&:session_id))` on line 240.

Reply:
Thanks, I will take a look.

Reply:
Addressed